src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -116,24 +124,20 @@ SELECT COUNT(DISTINCT p) FROM StudentParticipation p
 
     /**
      * @param exerciseId id of exercise
-     * @param correctionRound correction round to find completed assessments by
      * @return the number of completed assessments for the specified correction round of an exam exercise
      */
     @Query("""
-            SELECT COUNT(DISTINCT p)
-            FROM StudentParticipation p WHERE p.exercise.id = :exerciseId
-            AND p.testRun = FALSE
-            AND (SELECT COUNT(r)
-                FROM p.results r
-                WHERE r.assessor IS NOT NULL
-                AND r.rated = TRUE
-                AND r.submission = (select max(id) from p.submissions)
-                AND r.submission.submitted = TRUE
+            SELECT COUNT(p.id)
+            FROM StudentParticipation p left join  p.submissions s left join s.results r
+            WHERE p.exercise.id = :exerciseId
+                AND p.testRun = FALSE
+                AND s.submitted = TRUE
                 AND r.completionDate IS NOT NULL
-                AND (p.exercise.dueDate IS NULL OR r.submission.submissionDate <= p.exercise.dueDate)
-            ) >= (:correctionRound + 1L)
+                AND r.rated = TRUE
+                AND r.assessor IS NOT NULL
+                GROUP BY p.id
```

Comment:
I think this can just be p as we have overwritten .equal to use use the id anyway. Same in the count. just to count(p)

Reply:
I'm not sure here, at least in the count it didn't work with only p

Reply:
That is strange lol. But I guess then leave the id if it works

Reply:
It seems to me what you really want is the number of results per participation. So COUNT(DISTINCT r)) would make the most sense ,no?